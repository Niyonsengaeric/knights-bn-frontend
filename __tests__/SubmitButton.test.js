import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import ReactTestRenderer from 'react-test-renderer';
import SubmitButton from '../src/components/Button';

describe('Test the submit button ', () => {
  afterEach(cleanup);
  it('renders without crushing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SubmitButton />, div);
  });

  it('should render button correctly', () => {
    const { getByTestId } = render(<SubmitButton id="button" label="log in" />);
    expect(getByTestId('button')).toHaveTextContent('log in');
  });
  it('should render button correctly', () => {
    const { getByTestId } = render(
      <SubmitButton id="button" label="log out" />
    );
    expect(getByTestId('button')).toHaveTextContent('log out');
  });

  it('matches snapshot', () => {
    const tree = ReactTestRenderer.create(
      <SubmitButton id="button" label="log in" />
    );
    expect(tree).toMatchSnapshot();
  });
});
