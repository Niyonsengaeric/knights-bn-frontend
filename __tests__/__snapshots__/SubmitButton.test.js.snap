// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the submit button  matches snapshot 1`] = `
<button
  className="btn btn-primary"
  data-testid="button"
  disabled={false}
  id="button"
  label="log in"
  type="button"
>
  log in
</button>
`;
